File: ios/IDnowAutoIdent.m
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(IDnowAutoIdent, NSObject)

RCT_EXTERN_METHOD(startAutoIdent:(NSDictionary)a
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/IDnowAutoIdent.swift
import Foundation
import IDNowSDKCore

@objc(IDnowAutoIdent)
class IDnowAutoIdent: NSObject {

  @objc(startAutoIdent:withResolver:withRejecter:)
  func startAutoIdent(a: NSDictionary, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
      let language = a["language"] ?? "de"
      IDNowSDK.shared.start(token: a["id"], preferredLanguage: language, fromViewController: self, listener:{ (result: IDNowSDK.IdentResult, message: String) in
         if result == IDNowSDK.IdentResult.FINISHED {
             resolve("FINISHED")
         } else if result == IDNowSDK.IdentResult.CANCELLED {
             reject("CANCELLED")
         } else {
             reject("ERROR")
         }
      })
  }
}
